product_logistics_uom/tests/test_product_logistics_uom.py

```diff
@@ -0,0 +1,130 @@
+# Copyright 2023 ACSONE SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+
+
+from unittest.mock import patch
+
+from odoo.tests.common import TransactionCase
+
+
+class TestProductLogisticsUom(TransactionCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.env = cls.env(context=dict(cls.env.context, tracking_disable=True))
+        cls.product = cls.env["product.product"].create(
+            {
+                "name": "Test Product",
+            }
+        )
+        cls.weigh_uom_kg = cls.env.ref("uom.product_uom_kgm")
+        cls.weigh_uom_g = cls.env.ref("uom.product_uom_gram")
+        cls.volume_uom_m3 = cls.env.ref("uom.product_uom_cubic_meter")
+        cls.volume_uom_l = cls.env.ref("uom.product_uom_litre")
+        # set volume in m3 and weight in kg
+        cls.env["ir.config_parameter"].set_param("product.weight_in_lbs", "0")
+        cls.env["ir.config_parameter"].set_param("product.volume_in_cubic_feet", "0")
+
+    def test_product_volume(self):
+        self.product.volume_uom_id = self.volume_uom_l
+        self.product.volume = 1
+        self.assertEqual(self.product.product_volume, 1000)
+        self.product.product_volume = 10
+        self.assertEqual(self.product.volume, 0.01)
+        self.product.volume_uom_id = self.volume_uom_m3
+        self.assertEqual(self.product.product_volume, self.product.volume)
+
+    def test_product_show_product_uom_warning(self):
+        self.product.volume_uom_id = self.volume_uom_m3
+        self.product.volume = 0.0001
+        self.assertTrue(self.product.show_volume_uom_warning)
+        self.product.volume_uom_id = self.volume_uom_l
+        self.assertFalse(self.product.show_volume_uom_warning)
+
+    def test_product_weight(self):
+        self.product.weight_uom_id = self.weigh_uom_g
+        self.product.weight = 1
+        self.assertEqual(self.product.product_weight, 1000)
+        self.product.product_weight = 10
+        self.assertEqual(self.product.weight, 0.01)
+        self.product.weight_uom_id = self.weigh_uom_kg
+        self.assertEqual(self.product.product_weight, self.product.weight)
+
+    def test_product_show_product_weight_warning(self):
+        self.product.weight_uom_id = self.weigh_uom_kg
+        self.product.weight = 0.0001
+        self.assertTrue(self.product.show_weight_uom_warning)
+        self.product.weight_uom_id = self.weigh_uom_g
+        self.assertFalse(self.product.show_weight_uom_warning)
+
+    def test_template_volume(self):
+        template = self.product.product_tmpl_id
+        template.volume_uom_id = self.volume_uom_l
+        # Volume calculation from product_dimension module has compatibility issue.
+        with patch(
+            "odoo.addons.product_dimension.models.product_template.ProductTemplate._calc_volume",
```

Comment:
can you rebase into 18.0 from OCA to include `product_dimension` was merged before.
if not the test can be failed.
try to run this one: `odoo.sh tests product_logistics_uom 18.0`